Config.cpp

```diff
@@ -83,26 +84,59 @@ namespace {
             string result = parseStrFromLine("WINDOWED", line);
             ssConfig.Fullscreen = (result == "NO");
         }
+        int autoSegForXY = (int)std::ceil(std::sqrt(2) * (ssState.ScreenW + ssState.ScreenH) / TILEWIDTH);
         if (line.find("[SEGMENTSIZE_XY") != string::npos) {
-            int value = parseIntFromLine("SEGMENTSIZE_XY", line);
-            if (value < 1) {
-                value = DEFAULT_SIZE;
+            bool autosize = (parseStrFromLine("SEGMENTSIZE_XY", line) == "AUTO");
+            int value = DEFAULT_SEGSIZE_XY;
+            if (autosize) {
+                value = autoSegForXY;
+            } else {
+                value = parseIntFromLine("SEGMENTSIZE_XY", line);
+                value = ((value < MIN_SEGSIZE) ? DEFAULT_SEGSIZE_XY : value);
+                value = ((value > MAX_SEGSIZE) ? MAX_SEGSIZE : value);
             }
-            if (value > 100) {
-                value = 100;
+            //plus 2 to allow edge readings
+            value += 2;
```

Comment:
this block can be factored out into a function and shared by the three config elements that currently copy it.
```
ssState.Size.x = get_segment_size_config("SEGMENTSIZE_XY", line);
```